Fixed multiple choice issues and warnings

koertho opened the pull request:
This is another patch release for multiple choice and php 8 warnings. I tried many multiple choice combinations and hope I fixed the most glitches now.

cmette commented on src/Resources/contao/classes/SurveyQuestionMultiplechoice.php:
```diff
@@ -308,7 +306,7 @@ protected function calculateCumulated(): void
             } else {
                 if (!empty($arrAnswer['value'])) {
-                    ++$cumulated[$arrAnswer['value']];
+                    $cumulated[$arrAnswer['value']] = ($cumulated[$arrAnswer['value']] ?? 0) + 1;
```
Does this really work? My opinion is the following: If (under PHP 8+) a key (here $arrAnswer['value']) of an array (here $cumulated) is invalid, the expression $cumulated[$arrAnswer['value']] will never evaluate correctly to NULL. Thus, the right-hand side expression cannot be processed correctly under all circumstances and a warning may occur.

koertho replied:
In line 308 we make an empty check on $arrAnswer['value'], so there will never be an value where ?? would get an invalid value. But ?? will not check $arrAnswer['value'], only $comulated[...] in this line I think?

cmette replied:
yes, of course - you are right - I am only looking at 309 - not 308 :O(

cmette commented on src/Resources/contao/templates/survey/survey_question_multiplechoice.html5:
```diff
@@ -69,8 +69,8 @@
-        <?php if ($this->blnOther && isset($this->values['value'])): ?>
-            <div><input type="radio" name="<?php echo $this->ctrl_name; ?>" id="ctrl_<?php echo $this->ctrl_id; ?>_<?php echo $counter; ?>" class="radio<?php echo $this->ctrl_class; ?>" value="<?php echo $counter; ?>"<?php if ($this->values['value'] == $counter): ?> checked="checked"<?php endif; ?> /> <label for="ctrl_<?php echo $this->ctrl_id; ?>_<?php echo $counter; ?>"><?php echo $this->otherT ...
+        <?php if ($this->blnOther): ?>
+            <div><input type="radio" name="<?php echo $this->ctrl_name; ?>" id="ctrl_<?php echo $this->ctrl_id; ?>_<?php echo $counter; ?>" class="radio<?php echo $this->ctrl_class; ?>" value="<?php echo $counter; ?>"<?php if (isset($this->values['value']) && ($this->values['value'] == $counter)): ?> checked="checked"<?php endif; ?> /> <label for="ctrl_<?php echo $this->ctrl_id; ?>_<?php echo $co ...
```
If this template is rendered at the BE with act=list, then $this->values['value'] leads to an exception, because $this->values is undefined (here NULL). This is due to the loss of the widget context, which is not given at the BE in act=list. The problem would have to be solved globally.

koertho replied:
Oh, I just forgot an additional check here (i just fixed the first one and oversaw the second). If we let the check where it was before, we won't see any "other"-field output. This was the reason I changed it.

koertho replied:
Fixed in [96b12f2](https://github.com/pdir/contao-survey/pull/31/commits/96b12f2b2a39290b2237cdff84c0cc78dd872596)

cmette commented:
moin moin @koertho - I have left a few comments on the individual commits. From my point of view there is a central question: with which strategy can we move the code towards Contao 5? there are many calls to TL_MODE, the templates could be migrated to twig, Input:: could be replaced by access to the RequestStack and much more. I am aware that this is a challenging task.

koertho commented:
I also could reproduce issue in the backend with binäre question in results, so I can fix them

cmette commented:
Thanks for the TestDB and your reply.... I will talk to @MDevster. Then we see further, otherwise briefly about slack... Thanks a lot!

koertho commented:
Ok, I found the issue with binary question: In my PR for categories I changed the answer format for "normal" multiple choice answers to array, but not for binary answers. I changed/ fixed this (binary answers now also handled as array internal) in my last commit so we do not have to add additional check. I tested a lot in the backend and in the frontend and haven't found any issues. Would love to hear your result :)
